integr8ly/application-monitoring-operator
# INTLY-1100 Make inhibition rules more explicit for CronJob alerts

Currently the inhibition rule for CronJob alerts are vague and
encompass other alerts also. We only want to address the issue
where JobRunningTimeExceeded has a warning and a critical alert
for the same job.

This change restricts the inhibit_rule to just address alerts from
JobRunningTimeExceeded.

Verification:
- Deploy this application-monitoring-operator to a cluster
- Open up the AlertManager console, ensure the config in the Status
tab looks correct
- Ensure the backups playbook has been run on the cluster
  - Ensure their schedules are `*/1 * * * *` so you don't have to wait too long
  - CronJobs should exist
  - PrometheusRules should exist
- Edit the `backup-monitoring-alerts` PrometheusRule in the namespace:
  - For `JobRunningTimeExceeded` with `severity=warning`, change `300` to `120`
  - For `JobRunningTimeExceeded` with `severity=critical`, change `600` to `240`
- Pick a CronJob and edit it:
  - Set the value of `-d` (the line under it) in the command to `true`
- The CronJob should stay running forever, wait until the warning fires in
AlertManager. This should take about 2 minutes
- When the warning fires, edit a second CronJob the same way.
- Wait another 2 minutes
- The first CronJob alert should now be a `severity=critical`
- The second CronJob alert should be `severity=warning`
- If you click the show inhibited option you should see the inhibited warning
alert from the first CronJob.
